tests/v3_api/test_auth.py

```diff
@@ -147,13 +150,73 @@ def special_character_users_login(access_mode):
         disable_freeipa(admin_user, admin_token)
         enable_freeipa(admin_user, admin_token)
 
-    for user in auth_setup_data["specialcharacter_users"]:
-        allowed_principal_ids.append(principal_lookup(user, admin_token))
-    allowed_principal_ids.append(principal_lookup(admin_user, admin_token))
+    if AUTH_PROVIDER == "activeDirectory":
+        for user in auth_setup_data["specialchar_in_username"]:
+            allowed_principal_ids.append(principal_lookup(user, admin_token))
+        for user in auth_setup_data["specialchar_in_password"]:
+            allowed_principal_ids.append(principal_lookup(user, admin_token))
+        for user in auth_setup_data["specialchar_in_userdn"]:
+            allowed_principal_ids.append(principal_lookup(user, admin_token))
+        for group in auth_setup_data["specialchar_in_groupname"]:
+            for user in auth_setup_data[group]:
+                allowed_principal_ids.append(principal_lookup(user, admin_token))
 
-    add_users_to_siteAccess(admin_token, access_mode, allowed_principal_ids)
-    for user in auth_setup_data["specialcharacter_users"]:
-        login(user, PASSWORD)
+        allowed_principal_ids.append(principal_lookup(admin_user, admin_token))
+        add_users_to_siteAccess(admin_token, access_mode, allowed_principal_ids)
+
+        for user in auth_setup_data["specialchar_in_username"]:
+            login(user, PASSWORD)
+        for user in auth_setup_data["specialchar_in_password"]:
+            login(user, AD_SPECIAL_CHAR_PASSWORD)
+        for user in auth_setup_data["specialchar_in_userdn"]:
+            login(user, PASSWORD)
+        for group in auth_setup_data["specialchar_in_groupname"]:
+            for user in auth_setup_data[group]:
+                login(user, PASSWORD)
+
+    if AUTH_PROVIDER == "openLdap":
+        for user in auth_setup_data["specialchar_in_user_cn_sn"]:
+            allowed_principal_ids.append(principal_lookup(user, admin_token))
+        for user in auth_setup_data["specialchar_in_uid"]:
+            allowed_principal_ids.append(principal_lookup(user, admin_token))
+        for user in auth_setup_data["specialchar_in_password"]:
+            allowed_principal_ids.append(principal_lookup(user, admin_token))
+        for group in auth_setup_data["specialchar_in_groupname"]:
+            for user in auth_setup_data[group]:
+                allowed_principal_ids.append(principal_lookup(user, admin_token))
+
+        allowed_principal_ids.append(principal_lookup(admin_user, admin_token))
+        add_users_to_siteAccess(admin_token, access_mode, allowed_principal_ids)
```

Comment:
We should be adding the specialchar_in_groupname to the accustomed and then testing if all users belonging to this group have access